flang/test/Lower/allocatable-assignment.f90

```diff
@@ -12,280 +12,498 @@ module alloc_assign
+! character assignment ...
+
+! CHECK:  fir.if %[[VAL_8]]#0 {
+! CHECK:    fir.if %[[VAL_7]] {
+! CHECK:      fir.freemem %[[VAL_4]] : !fir.heap<!fir.char<1,?>>
+! CHECK:    }
+! CHECK:    %[[VAL_36:.*]] = fir.embox %[[VAL_8]]#1 typeparams %[[VAL_2]] : (!fir.heap<!fir.char<1,?>>, index) -> !fir.box<!fir.heap<!fir.char<1,?>>>
+! CHECK:    fir.store %[[VAL_36]] to %[[VAL_0]] : !fir.ref<!fir.box<!fir.heap<!fir.char<1,?>>>>
+! CHECK:  }
   x = "Hello world!"
 end subroutine
 
 ! CHECK-LABEL: func @_QMalloc_assignPtest_cst_char_scalar(
-! CHECK-SAME: %[[x:.*]]: !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>) {
+! CHECK-SAME:  %[[VAL_0:.*]]: !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>) {
 subroutine test_cst_char_scalar(x)
   character(10), allocatable  :: x
-  ! CHECK: fir.load %[[x]] : !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>
-  ! CHECK: fir.if %{{.*}} {
-  ! CHECK:   fir.if %false{{.*}} {
-  ! CHECK:   }
-  ! CHECK: } else {
-  ! CHECK:   %[[newAddr:.*]] = fir.allocmem !fir.char<1,10> {uniq_name = ".auto.alloc"}
-  ! CHECK:   %[[box:.*]] = fir.embox %[[newAddr]] : (!fir.heap<!fir.char<1,10>>) -> !fir.box<!fir.heap<!fir.char<1,10>>>
-  ! CHECK:   fir.store %[[box]] to %[[x]] : !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>
-  ! CHECK: }
-  ! CHECK: fir.load %[[x]] : !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>
+! CHECK:  %[[VAL_1:.*]] = constant 10 : index
+! CHECK:  %[[VAL_2:.*]] = fir.address_of(@_QQ{{.*}}) : !fir.ref<!fir.char<1,12>>
+! CHECK:  %[[VAL_3:.*]] = constant 12 : index
+! CHECK:  %[[VAL_4:.*]] = fir.load %[[VAL_0]] : !fir.ref<!fir.box<!fir.heap<!fir.char<1,10>>>>
+! CHECK:  %[[VAL_5:.*]] = fir.box_addr %[[VAL_4]] : (!fir.box<!fir.heap<!fir.char<1,10>>>) -> !fir.heap<!fir.char<1,10>>
+! CHECK:  %[[VAL_6:.*]] = fir.convert %[[VAL_5]] : (!fir.heap<!fir.char<1,10>>) -> i64
+! CHECK:  %[[VAL_7:.*]] = constant 0 : i64
+! CHECK:  %[[VAL_8:.*]] = cmpi ne, %[[VAL_6]], %[[VAL_7]] : i64
+! CHECK:  %[[VAL_9:.*]]:2 = fir.if %[[VAL_8]] -> (i1, !fir.heap<!fir.char<1,10>>) {
+! CHECK:    %[[VAL_10:.*]] = constant false
+! CHECK:    %[[VAL_11:.*]] = fir.if %[[VAL_10]] -> (!fir.heap<!fir.char<1,10>>) {
+! CHECK:      %[[VAL_12:.*]] = fir.allocmem !fir.char<1,10> {uniq_name = ".auto.alloc"}
+! CHECK:      fir.result %[[VAL_12]] : !fir.heap<!fir.char<1,10>>
+! CHECK:    } else {
+! CHECK:      fir.result %[[VAL_5]] : !fir.heap<!fir.char<1,10>>
+! CHECK:    }
+! CHECK:    fir.result %[[VAL_10]], %[[VAL_13:.*]] : i1, !fir.heap<!fir.char<1,10>>
+! CHECK:  } else {
+! CHECK:    %[[VAL_14:.*]] = constant true
+! CHECK:    %[[VAL_15:.*]] = fir.allocmem !fir.char<1,10> {uniq_name = ".auto.alloc"}
+! CHECK:    fir.result %[[VAL_14]], %[[VAL_15]] : i1, !fir.heap<!fir.char<1,10>>
+! CHECK:  }
+
+! character assignment ...
+! CHECK:  %[[VAL_24:.*]] = fir.convert %[[VAL_9]]#1 : (!fir.heap<!fir.char<1,10>>) -> !fir.ref<i8>
+! CHECK:  fir.call @llvm.memmove.p0i8.p0i8.i64(%[[VAL_24]], %{{.*}}, %{{.*}}, %{{.*}}) : (!fir.ref<i8>, !fir.ref<i8>, i64, i1) -> ()
```

Comment:
Just curious if these llvm intrinsic calls are not stable in lowering. Deltas due to the build compiler or something?